list-sensors.cpp

```diff
@@ -15,106 +15,216 @@
+class Properties : public PropertiesMap
 {
-    if (dbus_unit.compare(0, prefix_len, prefix) != 0)
+  public:
+    using PropertiesMap::PropertiesMap;
+
+    /**
+     * @brief Check is specified alarm property has true value
+     *
+     * @param name - Proprety name
+     */
+    bool isAlarmed(const PropertyName& name) const
     {
-        return "Unknown";
+        auto it = this->find(name);
+        return (it == this->end() ? false : std::get<bool>(it->second));
     }
 
-    std::string unit = dbus_unit.substr(prefix_len);
+    /**
+     * @brief Current sensor state
+     */
+    std::string status() const
+    {
+        std::string ret = "OK";
+        if (isAlarmed("CriticalAlarmLow") || isAlarmed("CriticalAlarmHigh"))
+        {
+            ret = "Critcal";
+        }
+        else if (isAlarmed("WarningAlarmLow") || isAlarmed("WarningAlarmHigh"))
+        {
+            ret = "Warning";
+        }
+
+        return ret;
+    }
 
-    if ("Volts" == unit)
+    /**
+     * @brief Sensor's scale factor
+     */
+    float scale() const
     {
-        return "V";
+        float ret = 1.f;
+        auto it = this->find("Scale");
+        if (it != this->end())
+        {
+            auto value = std::get<int64_t>(it->second);
+            ret = powf(10, static_cast<float>(value));
+        }
+        return ret;
     }
-    else if ("DegreesC" == unit)
+
+    /**
+     * @brief Format a sensor value or threshold
+     *
+     * @param name - Proprety name
+     *
+     * @return String with formated value of property
+     */
+    std::string get_prop_value(const PropertyName& name) const
     {
-        return "\u00B0C"; // UTF-8 Degrees symbol
+        std::string ret(9, '\0');
+        auto it = this->find(name);
+        if (it == this->end())
+        {
+            ret = "N/A";
+        }
+        else
+        {
+            auto _scale = scale();
+            auto _value = std::get<int64_t>(it->second);
+            if (_scale < 1.f)
+            {
+                snprintf(ret.data(), ret.size(), "%8.03f", _value * _scale);
+            }
+            else
+            {
+                snprintf(ret.data(), ret.size(), "%8d", (int)(_value * _scale));
+            }
+        }
+
+        return ret;
     }
-    else if ("Amperes" == unit)
+
+    std::string value() const
+    {
+        return get_prop_value("Value");
+    }
+    std::string criticalLow() const
     {
-        return "A";
+        return get_prop_value("CriticalLow");
     }
-    else if ("RPMS" == unit)
+    std::string criticalHigh() const
     {
-        return "RPM";
+        return get_prop_value("CriticalHigh");
     }
-    else if ("Watts" == unit)
+    std::string warningLow() const
     {
-        return "W";
+        return get_prop_value("CriticalLow");
     }
-    else if ("Joules" == unit)
+    std::string warningHigh() const
     {
-        return "J";
+        return get_prop_value("CriticalHigh");
     }
 
```

Comment:
The `Fatal` threshold is the `Vesnin` specific threshold which looks excess.
Besides, when any of sensors reach it the host is turned off immediately. It leads the sensors change its state and the `Fatal` state will never be shown or maybe will be shown during the very short time.

Do you think it still required here?

Reply:
First of all, you missed the main point of my comment. I would like a stronger incapsulation.

As for thresholds, do as we discussed internally.

Reply:
> First of all, you missed the main point of my comment. I would like a stronger incapsulation.

These two methods were written to avoid code duplicity. And yes, they look like macros. But I see nothing wrong with that as they can be called by class members only. I marked them as protected to emphasize it.

About the full C++ encapsulation which you mention earlier:
This will lead to some additional switch/case statements which convert an enum value to string.
It will look like an excess overhead in my opinion.

Reply:
Fatal threshold added.